byway/src/main/java/com/google/sps/servlets/DestinationsServlet.java

```diff
@@ -87,9 +90,42 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     }
     datastore.put(entity);
     ArrayList<String> destinations = (ArrayList<String>) entity.getProperty("destinations");
-    UserLocations userLocations = new UserLocations(start, destinations);    
+    Trip trip = new Trip(KeyFactory.keyToString(tripKey), start, destinations, new ArrayList<String>(), new ArrayList<String>());    
     response.setContentType("application/json;");
-    response.getWriter().println(gson.toJson(userLocations)); 
+    response.getWriter().println(gson.toJson(trip));
   }
-}
+  
 
+  /*
+  * Adds new user entity if user has not logged in before
+  * Adds tripKey to the user's array of trips
+  **/
+  public void addUserEntity(Key tripKey){
+    boolean userExists = false;
+    String userId = userService.getCurrentUser().getUserId();
+    String userEmail = userService.getCurrentUser().getEmail();
+    Query query = new Query("User");
+    PreparedQuery results = datastore.prepare(query);
+    for (Entity currentEntity : results.asIterable()) {
```

Comment:
the tripKey is for the Trip entity. This is a User entity. I have user entity set up so that it saves CurrentUser.getId() as a property, which is more reliable than emails since user emails can change.